lib/shared/thunder_popup_menu_item.dart

```diff
@@ -4,14 +4,22 @@ import 'package:flutter/material.dart';
 class ThunderPopupMenuItem extends PopupMenuItem {
   final IconData icon;
   final String title;
+  final bool? checkboxValue;
+  final void Function()? onCheckboxValueToggled;
 
-  ThunderPopupMenuItem({super.key, required super.onTap, required this.icon, required this.title})
-      : super(
+  ThunderPopupMenuItem({
+    super.key,
+    required super.onTap,
+    required this.icon,
+    required this.title,
+    this.checkboxValue,
+    this.onCheckboxValueToggled,
+  }) : super(
           child: ListTile(
-            dense: true,
-            horizontalTitleGap: 5,
-            leading: Icon(icon, size: 20),
-            title: Text(title),
-          ),
+              dense: true,
+              horizontalTitleGap: 5,
+              leading: Icon(icon, size: 20),
+              title: Text(title),
+              trailing: checkboxValue == null ? null : Checkbox(value: checkboxValue, onChanged: (_) => onCheckboxValueToggled?.call())),

```

Comment:
I think it might be better to pass in a `trailing` widget rather than `checkboxValue` and `onCheckboxValueToggled` into ThunderPopupMenuItem.

Doing this will allow us to place any arbitrary widget (e.g., icons, buttons, etc.) as part of the trailing end of this ListTile!

Reply:
Good idea, done! In addition, I removed the usage of `Checkbox` and just passed an `Icon`.